### ISSUE TITLE
Managed plugin does not work on PhotonOS (mount fails with "wrong args") 

### ISSUE THREAD

**Issue by msterin** [vmware/docker-volume-vsphere#1033]
It give "mount error: wrong arguments" , it feels that the block device (/dev/py-path...) does not exist yet when the mount is tried. Also, I saw timeout  error message from inotify wait.

For 1.13, we need to document it and open a bug against PhotonOS I guess since it works fine on Ubuntu 

//CC @tusharnt

**Comment by govint** [comment]
Is this on Nimbus or on physical host. the notify uses a 1 second timeout
and this timeout happens most of the time on Ubuntu as well.

**Comment by msterin** [comment]
Nope - fusion with ESX on mac

**Comment by ashahi1** [comment]
Got the ova from: https://vmware.github.io/photon/

Tried with both Photon OS, Version 1.0 — OVA with virtual hardware v10 as well as virtual hardware v11 but did not see any issue in mounting the volume to container. Everything worked fine.

Test Steps:
1. Downloaded the docker-volume-vsphere managed plugin 
2. Create a volume using the managed plugin
3. Ran a container with volume mounted
4. Exited the container and removed the volumes


Setup details are as follows:

```
root@photon-XT7BYSpoc [ ~ ]# uname -a
Linux photon-XT7BYSpoc 4.4.8-esx #1-photon SMP Tue Jun 7 08:04:49 UTC 2016 x86_64 GNU/Linux
root@photon-XT7BYSpoc [ ~ ]#

Docker Version:      17.03.0-ce
```

---------

Test steps and their output with Photon OS, Version 1.0 — OVA with virtual hardware v10 is as follows:


1. No plugins were there and no volumes were there on the setup

```
root@photon-XT7BYSpoc [ ~ ]# docker plugin ls
ID                  NAME                DESCRIPTION         ENABLED
root@photon-XT7BYSpoc [ ~ ]# docker volume ls
DRIVER              VOLUME NAME
```

2. Installed the plugin

```
root@photon-XT7BYSpoc [ ~ ]# docker plugin install cnastorage/docker-volume-vsphere:1.0
Plugin "cnastorage/docker-volume-vsphere:1.0" is requesting the following privileges:
 - network: [none]
 - mount: [/dev]
 - mount: [/var/run]
 - mount: [/etc]
 - mount: [/var/log]
 - allow-all-devices: [true]
 - capabilities: [CAP_SYS_ADMIN]
Do you grant the above permissions? [y/N] y
1.0: Pulling from cnastorage/docker-volume-vsphere
2ba055944850: Download complete
Digest: sha256:aaa7ccee39b8ba5ae16465d60069469b0141057a839cc1d14b22ae0008f4902b
Status: Downloaded newer image for cnastorage/docker-volume-vsphere:1.0
Installed plugin cnastorage/docker-volume-vsphere:1.0
root@photon-XT7BYSpoc [ ~ ]# docker plugin ls
ID                  NAME                                   DESCRIPTION                           ENABLED
83f731f2223b        cnastorage/docker-volume-vsphere:1.0   VMWare vSphere Docker Volume plugin   true
```

3. Created a volume using the plugin

```
root@photon-XT7BYSpoc [ ~ ]# docker volume create -d cnastorage/docker-volume-vsphere:1.0 --name=TestVol1
TestVol1
root@photon-XT7BYSpoc [ ~ ]# 

docker volume ls
DRIVER                                 VOLUME NAME
cnastorage/docker-volume-vsphere:1.0   TestVol1@sharedVmfs-0

```

4. Ran the container with volume mounted and also wrote data to volume
```
root@photon-XT7BYSpoc [ ~ ]# docker run -it --volume-driver=cnastorage/docker-volume-vsphere:1.0 -v TestVol1@sharedVmfs-0:/vol1 --name ub ubuntu
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu
d54efb8db41d: Pull complete
f8b845f45a87: Pull complete
e8db7bf7c39f: Pull complete
9654c40e9079: Pull complete
6d9ef359eaaa: Pull complete
Digest: sha256:dd7808d8792c9841d0b460122f1acf0a2dd1f56404f8d1e56298048885e45535
Status: Downloaded newer image for ubuntu:latest
root@5de6e0b464f2:/# df -h
Filesystem                                       Size  Used Avail Use% Mounted on
overlay                                          7.8G  743M  6.7G  10% /
tmpfs                                           1003M     0 1003M   0% /dev
tmpfs                                           1003M     0 1003M   0% /sys/fs/cgroup
/dev/disk/by-path/pci-0000:0b:00.0-scsi-0:0:0:0   93M  1.6M   85M   2% /vol1
/dev/root                                        7.8G  743M  6.7G  10% /etc/hosts
shm                                               64M     0   64M   0% /dev/shm
tmpfs                                           1003M     0 1003M   0% /sys/firmware
root@5de6e0b464f2:/# echo "I'm vol1 from photon vm" > vol1/hellos
root@5de6e0b464f2:/# cat vol1/hellos
I'm vol1 from photon vm
root@5de6e0b464f2:/#

```

Admin cli volume ls output:

```
[root@sc-rdops-vm02-dhcp-50-83:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Volume    Datastore     VM-Group  Capacity  Used  Filesystem  Policy  Disk Format  Attached-to  Access      Attach-as               Created By  Created Date
--------  ------------  --------  --------  ----  ----------  ------  -----------  -----------  ----------  ----------------------  ----------  ------------------------
TestVol1  sharedVmfs-0  _DEFAULT  100MB     15MB  ext4        N/A     thin         Photon       read-write  independent_persistent  Photon      Tue Mar 21 18:30:12 2017

[root@sc-rdops-vm02-dhcp-50-83:~]
```

5. Exited the container and removed the volume

```
root@photon-XT7BYSpoc [ ~ ]# docker volume ls
DRIVER                                 VOLUME NAME
cnastorage/docker-volume-vsphere:1.0   TestVol1@sharedVmfs-0
root@photon-XT7BYSpoc [ ~ ]# docker volume rm TestVol1
TestVol1
root@photon-XT7BYSpoc [ ~ ]# docker volume ls
DRIVER              VOLUME NAME
root@photon-XT7BYSpoc [ ~ ]#

```


=========================================================

Test steps and their output with Photon-1.0 OVA with virtual hardware v11

```

root@photon-nH5qcjVR6 [ ~ ]# docker plugin ls
ID                  NAME                DESCRIPTION         ENABLED
root@photon-nH5qcjVR6 [ ~ ]# docker volume ls
DRIVER              VOLUME NAME
root@photon-nH5qcjVR6 [ ~ ]# docker plugin install cnastorage/docker-volume-vsphere:1.0
Plugin "cnastorage/docker-volume-vsphere:1.0" is requesting the following privileges:
 - network: [none]
 - mount: [/dev]
 - mount: [/var/run]
 - mount: [/etc]
 - mount: [/var/log]
 - allow-all-devices: [true]
 - capabilities: [CAP_SYS_ADMIN]
Do you grant the above permissions? [y/N] y
1.0: Pulling from cnastorage/docker-volume-vsphere
2ba055944850: Download complete
Digest: sha256:aaa7ccee39b8ba5ae16465d60069469b0141057a839cc1d14b22ae0008f4902b
Status: Downloaded newer image for cnastorage/docker-volume-vsphere:1.0
Installed plugin cnastorage/docker-volume-vsphere:1.0
root@photon-nH5qcjVR6 [ ~ ]# docker plugin ls
ID                  NAME                                   DESCRIPTION                           ENABLED
03037a5dc97e        cnastorage/docker-volume-vsphere:1.0   VMWare vSphere Docker Volume plugin   true
root@photon-nH5qcjVR6 [ ~ ]# docker volume create -d cnastorage/docker-volume-vsphere:1.0 --name=TestVol11
TestVol11
root@photon-nH5qcjVR6 [ ~ ]# docker volume ls
DRIVER                                 VOLUME NAME
cnastorage/docker-volume-vsphere:1.0   TestVol11@sharedVmfs-0
root@photon-nH5qcjVR6 [ ~ ]# docker run -it --volume-driver=cnastorage/docker-volume-vsphere:1.0 -v TestVol11@sharedVmfs-0:/vol1 --name ub                                                            ubuntu
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu
d54efb8db41d: Pull complete
f8b845f45a87: Pull complete
e8db7bf7c39f: Pull complete
9654c40e9079: Pull complete
6d9ef359eaaa: Pull complete
Digest: sha256:dd7808d8792c9841d0b460122f1acf0a2dd1f56404f8d1e56298048885e45535
Status: Downloaded newer image for ubuntu:latest
root@ff419cd0b5e7:/# df -h
Filesystem                                       Size  Used Avail Use% Mounted on
overlay                                          7.8G  743M  6.7G  10% /
tmpfs                                           1003M     0 1003M   0% /dev
tmpfs                                           1003M     0 1003M   0% /sys/fs/cgroup
/dev/disk/by-path/pci-0000:0b:00.0-scsi-0:0:0:0   93M  1.6M   85M   2% /vol1
/dev/root                                        7.8G  743M  6.7G  10% /etc/hosts
shm                                               64M     0   64M   0% /dev/shm
tmpfs                                           1003M     0 1003M   0% /sys/firmware
root@ff419cd0b5e7:/# echo "I'm vol1 from photon vm-11" > vol1/hellos
root@ff419cd0b5e7:/# cat vol1/hellos
I'm vol1 from photon vm-11
root@ff419cd0b5e7:/#



[root@sc-rdops-vm02-dhcp-50-83:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Volume     Datastore     VM-Group  Capacity  Used  Filesystem  Policy  Disk Format  Attached-to   Access      Attach-as              Created By    Created Date                                      
---------  ------------  --------  --------  ----  ----------  ------  -----------  ------------  ----------  ---------------------  ------------  --------------------                              
TestVol11  sharedVmfs-0  _DEFAULT  100MB     15MB  ext4        N/A     thin         Photon-HW-11  read-write  independent_persist..  Photon-HW-11  Tue Mar 21 19:07:5..                              

[root@sc-rdops-vm02-dhcp-50-83:~]




root@photon-nH5qcjVR6 [ ~ ]# docker rm ub
ub
root@photon-nH5qcjVR6 [ ~ ]# docker volume ls
DRIVER                                 VOLUME NAME
cnastorage/docker-volume-vsphere:1.0   TestVol11@sharedVmfs-0
root@photon-nH5qcjVR6 [ ~ ]# docker volume rm TestVol11@sharedVmfs-0
TestVol11@sharedVmfs-0
root@photon-nH5qcjVR6 [ ~ ]#



[root@sc-rdops-vm02-dhcp-50-83:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Volume  Datastore  VM-Group  Capacity  Used  Filesystem  Policy  Disk Format  Attached-to  Access  Attach-as  Created By  Created Date
------  ---------  --------  --------  ----  ----------  ------  -----------  -----------  ------  ---------  ----------  ------------

[root@sc-rdops-vm02-dhcp-50-83:~]

**Comment by ashahi1** [comment]
@msterin Please take a look at the steps and their output posted above. Please update if I am missing some steps or this issue can be closed. 

Thanks!

**Comment by msterin** [comment]
Looks great. I guess this was another instance of #962 (fixed in #1058). Feel free to close.

**Comment by ashahi1** [comment]
@msterin  Thanks! Closing this issue.
